File: 242-valid-anagram/valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length !== t.length) return false

    const counter = {}

    for (let char of s) {
        counter[char] = counter[char] ? counter[char] += 1 : 1
    }

    for (let char of t) {
        if (!counter[char] || counter[char] === 0) {
            return false
        }

        counter[char] -= 1
    }

    return true
};

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        char_count = {}

        for char in s:
            if char in char_count:
                char_count[char] += 1
            else:
                char_count[char] = 1

        for char in t:
            if char_count.get(char, 0) <= 0:
                return False
            char_count[char] -= 1

        return max(char_count.values()) == 0